psycopg/utils.c

```diff
@@ -50,8 +50,13 @@ psycopg_escape_string(connectionObject *conn, const char *from, Py_ssize_t len,
     Py_ssize_t ql;
     int eq = (conn && (conn->equote)) ? 1 : 0;
 
-    if (len == 0)
+    if (len == 0) {
         len = strlen(from);
+    } else if (strcspn(from, "\0") != len) {
```

Comment:
Actually `stren(from) == len` would be a nice trick, but sure, would deserve a line of comment. :D

Actually^2, I don't even get how the `strcspn` implementation can even work using `reject = "\0"`: `strlen(reject)` is 0, not 1, isn't it? I'm not that sure that this use is portable, for some reasonable implementation of `strcspn`.

Any `strcspn` implementation will be more complicated that `strchr` and here we are in a busy loop of the adapter, so I actually prefer to use the more conceptually simple `strchr` and a more "complex" final test. Note however that `man strchr(3)` says that 

> The strchr() and strrchr() functions return a pointer to the matched character or NULL if the character is  not  found.
> The terminating null byte is considered part of the string, so that if c is specified as '\0', these functions return a
> pointer to the terminator.

so ISTM `strchr` will never return NULL using `'\0'` as input char and we only need `if (strchr(from, '\0') != len) { // exception` as a test.